### ISSUE TITLE
LAPOR implementation is not complete 

### ISSUE THREAD

**Issue by JonasOberhauser** [MPI-SWS/genmc#11]
Generates one execution without lapor, 0 with -lapor

```c
#include <pthread.h>
#include <assert.h>

#include <stdatomic.h>

#define NTHREADS 2



#include <stdint.h>
#include <stdio.h>



static atomic_int x = 0, y = 0;
static pthread_mutex_t mylockA,mylockB,mylockC;

#define CS {                                        \
        iter++;                                     \
        int t = x+1;                                \
        /*printf("%dT %ld: t = %d\n", iter, id, t); */  \
        x = t;                                      \
    }


static void *run(void *arg)
{
    intptr_t id = (intptr_t)arg;
    int iter = 0;
    pthread_mutex_lock(&mylockA);
    CS
    if (id == 1)
        pthread_mutex_unlock(&mylockA);
    pthread_mutex_lock(&mylockB);
    CS
    if (id == 0)
        pthread_mutex_unlock(&mylockB);    

    return NULL;
}

int main()
{
        pthread_t t[NTHREADS];
    //printf("===\n");

        for (intptr_t i = 0; i < NTHREADS; i++)
                pthread_create(t+i, 0, run, (void*)i);
        for (intptr_t i = 0; i < NTHREADS; i++)
                pthread_join(t[i], NULL);


        return 0;
}
```

**Comment by michaliskok** [comment]
That's a good observation. However, the executions of these programs are not lock-well-formed. According to the paper, LAPOR should support an error in such cases, although the implementation does not do that in practice. I think the reason I had in mind when implementing the code was to accommodate `assume()` statements within critical sections, as such constructs can easily lead to not lock-well-formed executions. 

Perhaps printing at least a warning would be a better option.

**Comment by JonasOberhauser** [comment]
I looked into the paper before reporting the issue, but couldn't find that constraint at the time (only that the memory model needs to be lock-extensible). I guess this sentence confused me: "That is, if the underlying memory model is well-formed,
prefix-closed and lock-extensible, then LAPOR is complete."

**Comment by michaliskok** [comment]
GenMC does print a warning since v0.6(?), and some spurious warnings have been fixed in v0.6.1.
